certbot_dns_cpanel/installer.py

```diff
@@ -0,0 +1,187 @@
+        dnsauth = dns_common.DNSAuthenticator(self.config, __name__)
+        dnsauth.dest = self.dest
+        self.credentials = dnsauth._configure_credentials(
+            'credentials',
+            'The cPanel credentials INI file',
+            None,
+            self._validate_credentials
+        )
+        
+        if not self.credentials:
+            raise errors.Error('No auth data')
+
+        if self.credentials.conf('token'):
+            return _CPanelClient(self.credentials.conf('url'), self.credentials.conf('username'), None, self.credentials.conf('token'))
+        elif self.credentials.conf('password'):
+            return _CPanelClient(self.credentials.conf('url'), self.credentials.conf('username'), self.credentials.conf('password'), None)
+
+        return _CPanelClient(
+            self.credentials.conf('url'),
+            self.credentials.conf('username'),
+            self.credentials.conf('password'),
+            self.credentials.conf('token'),
+        )
+
+    def deploy_cert(self, domain, cert_path, key_path, chain_path, fullchain_path):
+        self._get_cpanel_client().deploy_cert(domain, cert_path, key_path, chain_path, fullchain_path, self)
+
+    def enhance(self, domain, enhancement, options=None):  # pylint: disable=missing-docstring,no-self-use
+        pass  # pragma: no cover
+
+    def supported_enhancements(self):  # pylint: disable=missing-docstring,no-self-use
+        return []  # pragma: no cover
+
+    def get_all_certs_keys(self):  # pylint: disable=missing-docstring,no-self-use
+        pass  # pragma: no cover
+
+    def save(self, title=None, temporary=False):  # pylint: disable=no-self-use
+	# this could be useful to locally keep a list of certificates installed and then clean up old ones for the same domain on servers
+	# however cpanel does not seem to have an api to delete certificates
+	# and, cpanel seems to not duplicate if the same certificate is installed
+        pass  # pragma: no cover
+
+    def rollback_checkpoints(self, rollback=1):  # pylint: disable=missing-docstring,no-self-use
+        pass  # pragma: no cover
+
+    def recovery_routine(self):  # pylint: disable=missing-docstring,no-self-use
+        pass  # pragma: no cover
+
+    def view_config_changes(self):  # pylint: disable=missing-docstring,no-self-use
+        pass  # pragma: no cover
+
+    def config_test(self):  # pylint: disable=missing-docstring,no-self-use
+        pass  # pragma: no cover
+
+    def restart(self):  # pylint: disable=missing-docstring,no-self-use
+        pass  # pragma: no cover
+
+    def renew_deploy(self, lineage, *args, **kwargs): # pylint: disable=missing-docstring,no-self-use
+        """
+        Renew certificates when calling `certbot renew`
+        """
+        # Run deploy_cert with the lineage params
+        self.deploy_cert(lineage.names()[0], lineage.cert_path, lineage.key_path, lineage.chain_path, lineage.fullchain_path)
+
+        return
+
+class _CPanelClient:
```

Comment:
Instead of duplicating code between Authenticator and Installer, it would be better to have everything in a same "common" class for communicating with cpanel.